### PR TITLE
add SSL functionality

### PR DESCRIPTION
**Pull request by J3rr1ck**
hey looks like my setup was working, matches your code style pretty well also.

### CODE REVIEW

**Comment by J3rr1ck** [comment]
image
image for proof on function

File: diffuzers/cli/run_app.py [hunk 1]
```diff
@@ -106,6 +122,10 @@ def run(self):
             "--",
             "--device",
             self.device,
+            "--ssl-certfile",
```
Comment (abhishekkrthakur) [inline diffuzers/cli/run_app.py:125]: these need to be optional params and must be added to list only when available

File: diffuzers/cli/run_app.py [hunk 2]
```diff
@@ -106,6 +122,10 @@ def run(self):
...
+            self.ssl_certfile,
```
Comment (abhishekkrthakur) [inline diffuzers/cli/run_app.py:126]: does it even work with streamlit? :)

**Review by abhishekkrthakur — APPROVED** [review]

**Review by abhishekkrthakur — CHANGES_REQUESTED** [review]
convert to optional args when passing to subprocess

File: diffuzers/cli/run_app.py (hunk 2 above)
Comment (J3rr1ck) [reply diffuzers/cli/run_app.py:126]: Crap honestly totally forgot to test with streamlit, need to refer to their docs for a few things today anywho 
Can follow back up with a few fixes such as tls reqs and fixes for atreamlit

**Comment by J3rr1ck** [comment]
Confirming streamlit doesn't have any support for ssl, will need to write a middle layer for handling that which I won't have time for till end of week (yay dayjobs!) removed gui ssl implementation till I (or anyone else) wants to smash through some extra coding.

File: diffuzers/cli/run_app.py [hunk 3]
```diff
@@ -62,15 +62,18 @@ def register_subcommand(parser: ArgumentParser):
             required=False,
             help="Ngrok key to use for sharing the app. Only required if you want to share the app",
         )
+        
         run_app_parser.set_defaults(func=run_app_command_factory)
 
-    def __init__(self, output, share, port, host, device, ngrok_key):
+    def __init__(self, output, share, port, host, device, ngrok_key, ssl_certfile, ssl_keyfile):
```
Comment (abhishekkrthakur) [inline diffuzers/cli/run_app.py:68]: this needs removal too :)

**Review by abhishekkrthakur — CHANGES_REQUESTED** [review]

**Comment by abhishekkrthakur** [comment]
No worries. Just a minor thing left in run_app.py

File: diffuzers/cli/run_app.py [hunk 4]
```diff
@@ -71,6 +72,8 @@ def __init__(self, output, share, port, host, device, ngrok_key):
         self.host = host
         self.device = device
         self.ngrok_key = ngrok_key
+        self.ssl_certfile = ssl_certfile
```
Comment (abhishekkrthakur) [inline diffuzers/cli/run_app.py:75]: sorry to bother you again, but there's still a couple of lines left

**Review by abhishekkrthakur — CHANGES_REQUESTED** [review]

**Pull request merged**